drivers/sensor/nordic/qdec_nrfx/qdec_nrfx.c

```diff
@@ -173,81 +174,81 @@ static const struct sensor_driver_api qdec_nrfx_driver_api = {
 	.trigger_set  = qdec_nrfx_trigger_set,
 };
 
-#ifdef CONFIG_PM_DEVICE
-static int qdec_nrfx_pm_action(const struct device *dev,
-			       enum pm_device_action action)
+static int qdec_pm_suspend(const struct device *dev, bool disable)
 {
 	const struct qdec_nrfx_config *config = dev->config;
-	int ret = 0;
+
+	if (disable) {
+		nrfx_qdec_disable(&config->qdec);
+	}
+
+	qdec_nrfx_gpio_ctrl(dev, false);
+	return pinctrl_apply_state(config->pcfg, PINCTRL_STATE_SLEEP);
+}
+
+static int qdec_pm_resume(const struct device *dev)
+{
+	const struct qdec_nrfx_config *config = dev->config;
+	int ret;
+
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0) {
+		return ret;
+	}
+
+	qdec_nrfx_gpio_ctrl(dev, true);
+	nrfx_qdec_enable(&config->qdec);
+
+	return 0;
+}
+
+static int qdec_pm_turn_on(const struct device *dev)
+{
+	const struct qdec_nrfx_config *config = dev->config;
+	nrfx_err_t nerr;
+
+	nerr = nrfx_qdec_init(&config->qdec, &config->config, qdec_nrfx_event_handler, (void *)dev);
+	if (nerr == NRFX_SUCCESS) {
+		return qdec_pm_suspend(dev, false);
+	}
+
+	return (nerr == NRFX_ERROR_INVALID_STATE) ? -EBUSY : -EFAULT;
+}
+
+static int qdec_nrfx_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	int ret = -ENOTSUP;
 
 	switch (action) {
-	case PM_DEVICE_ACTION_RESUME:
-		ret = pinctrl_apply_state(config->pcfg,
-					  PINCTRL_STATE_DEFAULT);
-		if (ret < 0) {
-			return ret;
-		}
-		qdec_nrfx_gpio_ctrl(dev, true);
-		nrfx_qdec_enable(&config->qdec);
+	case PM_DEVICE_ACTION_TURN_ON:
+		ret = qdec_pm_turn_on(dev);
```

Comment:
does this device really support being turned on/off?

Reply:
yeah, rather not. I've moved this part back to init.